### PR TITLE
semantic-release: Add missing fields in NextRelease

### PR DESCRIPTION
**Pull request by sebbo2002**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
    - **Keys in NextRelease**: https://semantic-release.gitbook.io/semantic-release/developer-guide/plugin#verifyrelease
    - **Possible values for type**: https://semantic-release.gitbook.io/semantic-release/developer-guide/plugin#multiple-analyzecommits-plugins
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/semantic-release/index.d.ts [hunk 1]
```diff
@@ -8,6 +8,23 @@
...
+     * is possible to create more advanced rules or situations, e.g. if
+     * none of the commits would result in new release, then a default
+     * can be defined.
+     *
+     * For this reason, the types of semantic-release's default analyzer,
+     * commit-analyzer, are noted here.
+     *
+     * See [Plugin development](https://semantic-release.gitbook.io/semantic-release/developer-guide/plugin#multiple-analyzecommits-plugins)
+     * for more details.
+     */
```
Comment (djcsdy) [inline types/semantic-release/index.d.ts:25]: This documentation doesn't describe the type being documented. I can see it has been copied from the semantic-release documentation, but out of context it doesn't make sense to me.

I think this documentation should say something like "A semver release type. See https://github.com/semantic-release/commit-analyzer/blob/master/lib/default-release-types.js"

File: types/semantic-release/semantic-release-tests.ts [hunk 2]
```diff
@@ -166,5 +166,14 @@ const result3: lib.Result = {
         gitTag: "v1.2.0",
         notes: "",
         pluginName: "@semantic"
+    }, {
+        name: "example-lib",
+        url: "https://www.npmjs.com/package/example-lib",
+        type: "prerelease",
+        version: "1.2.0-pre.2",
+        gitHead: "a018aff59995a17c0564fa3fd0cb96223f4d4096",
+        gitTag: "1.2.0-pre.2",
```
Comment (djcsdy) [inline types/semantic-release/semantic-release-tests.ts:175]: Should be `v1.2.0-pre.2` (missing `v`).

**Review by djcsdy — CHANGES_REQUESTED** [review]
Looks good except for minor issues noted

**Review by djcsdy — APPROVED** [review]

**Comment by sebbo2002** [comment]
@typescript-bot Ready to merge

**Pull request merged**